edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/IModuleIdentity.cs

```diff
@@ -1,11 +1,15 @@
 // Copyright (c) Microsoft. All rights reserved.
 namespace Microsoft.Azure.Devices.Edge.Agent.Core
 {
+    using Microsoft.Azure.Devices.Edge.Util;
+
     public interface IModuleIdentity
     {
         string IotHubHostname { get; }
 
-        string GatewayHostname { get; }
+        Option<string> ParentEdgeHostname { get; }
+
+        string EdgeDeviceHostname { get; }
```

Comment:
I wonder if it is worth plumbing this information through the identity. 
The Command factory has access to the config. Maybe we can just get from that?

Reply:
From my understanding, parent edge is part of the identity information for a device/module.  Do you have specific concern if parent edge hostname is added here?